indigo_api/models/works.py

```diff
@@ -462,14 +462,30 @@ class Meta:
         unique_together = (('commenced_work', 'commencing_work', 'date'),)
 
     def save(self, *args, **kwargs):
-        # ensure only one commencement with main=True on commenced work
-        existing_main = self.commenced_work.main_commencement
+        work = self.commenced_work
+        # ensure only one commencement with main=True exists on the commenced work
+        existing_main = work.main_commencement
         if existing_main and existing_main != self:
             self.main = False
 
-        # ensure only one commencement with all_provisions=True on commenced work
-        existing_all = self.commenced_work.commencements.filter(all_provisions=True).first()
-        if existing_all and existing_all != self:
+        # if more than one commencement / uncommencement exists on the work, all_provisions is False on all of them
+        other_commencements = work.commencements.exclude(pk=self.pk)
+        uncommenced = work.uncommenced_provisions
+
+        if other_commencements or uncommenced:
+            self.all_provisions = False
+
+        for commencement in other_commencements:
```

Comment:
On second thought, I think we should move this logic into the view where we update a commencement object. Otherwise we're going to be chasing our tails, as each `.save()` triggers this logic again and again.

Perhaps as a rule of thumb, `.save()` should only adjust the current object. Adjusting other objects should happen at a higher level, where the code has a better "view" of all the changes being made to a commencement object.